modules/admin-ui/src/main/java/org/opencastproject/adminui/endpoint/FeedEndpoint.java

```diff
@@ -0,0 +1,65 @@
+/**
+ * Licensed to The Apereo Foundation under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ *
+ * The Apereo Foundation licenses this file to you under the Educational
+ * Community License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License
+ * at:
+ *
+ *   http://opensource.org/licenses/ecl2.txt
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+
+package org.opencastproject.adminui.endpoint;
+
+import org.opencastproject.serviceregistry.api.RemoteBase;
+import org.opencastproject.util.doc.rest.RestService;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.methods.HttpGet;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+@Path("/")
+@RestService(name = "FeedService", title = "Admin UI Feed Service",
+  abstractText = "Provides Feed Information",
+  notes = {"This service offers Feed information for the admin UI."})
+public class FeedEndpoint extends RemoteBase {
+
+  public FeedEndpoint() {
+    super("org.opencastproject.feed.impl.FeedServiceImpl");
+  }
+
+  @GET
+  @Path("/listFeedServices")
+  @Produces(MediaType.APPLICATION_JSON)
+  public String listFeedServices() {
+
+    HttpGet get = new HttpGet("/listFeedServices");
```

Comment:
I would have rather expected a method like FeedGenerator[] getFeedGenerators() in the FeedServiceImpl. The FeedEndpoint (respectively the module AdminUI) would then require a dependency to the FeedService and use that method here to retrieve the feed generators (as Java objects) and create the response (JSON).

That why you also could avoid the weird mixup of XML and JSON in the FeedServiceImpl.

Reply:
`FeedServiceImpl` isn't necessarily available on the same node as the admin interface. It usually resides on the presentation node.

Reply:
the current URL is /feeds/listFeedServices, I could change it to /feeds/admin-ui/feeds but this looks weird, probably /feeds/admin-ui, but I don't know If I can change the root "feeds" to something else, it seems that "feeds" is fix

Reply:
The URL `/admin-ui/feeds` comes from https://github.com/opencast/opencast/pull/845/files#diff-fbcfee710538a23c249dab87b06f3769R8 and is fine.

What I meant is to directly use this URL as `GET /admin-ui/feeds` already suggests that it retrieves the list of feeds. Looking at other similar requests in the Admin UI Facade, e.g. `GET /admin-ui/series/series.json` (see https://github.com/opencast/opencast/blob/develop/modules/admin-ui/src/main/java/org/opencastproject/adminui/endpoint/SeriesEndpoint.java#L536), it seems that `GET /admin-ui/feeds/feeds.json` would be most consistent with other list retrieval requests in the Admin UI.

Reply:
I changed it to:
/feeds/feeds.json (backend)
/admin-ng/feeds/feeds.json (frontend -> req to backend in another branch)

I hope that this is okay